### DISCUSSION TITLE
error: SEH exception with code 0xc0000005 thrown in the test body.vips_avg()/vips_deviate()

### DISCUSSION THREAD

**Discussion by inloveable** [libvips/libvips#3435] [Q&A]
I'm tring to use libvips to compute two properties of my image.

evering things works fine untill trying to invoke vips_avg()，an read acess violation had beeng triggerd.

my source code is under below,what might be the issue? Is it somethine I miss from the document?Thank you.

![微信图片_20230412000051](https://user-images.githubusercontent.com/88480407/231221575-61ccb131-e7c2-4a5b-8ea7-4566b733f474.png)

**Comment by jcupitt** [comment]
Hi again,

I would use the C++ API, it's much simpler:

```C++
VImage image = VImage::new_from_file(argv[1]).colourspace("b-w");
double mean = image.avg();
double stddev = image.deviate();
```

**Reply by inloveable** [reply]
It's quite hard for me to use c++ api(I really wish I could,c api is driving me crazy.)

I'm just wondering is there any solution to this?I'm simply tring to find out whether a picture is a background in a large pyramid(which was cut using dzsave)

Thank you so much for your time.

**Reply by jcupitt** [reply]
On windows, add a project to your VS solution for the libvips C++ API using these source files:

https://github.com/libvips/libvips/tree/master/cplusplus

Then in your project, add that project as a dependency.

**Comment by jcupitt** [comment]
If you must use the C API, it's something like:

```C
/* compile with:
 *
 *      gcc -g -Wall stats.c `pkg-config vips --cflags --libs`
 */

#include <vips/vips.h>

int
main( int argc, char **argv )
{
    VipsImage *image;
    VipsImage *x;

    double mean;
    double stddev;

    VIPS_INIT( argv[0] );

    if( !(image = vips_image_new_from_file( argv[1], NULL )) ) 
        vips_error_exit( NULL ); 

    if( vips_colourspace( image, &x, VIPS_INTERPRETATION_B_W, NULL ) ) {
        g_object_unref( image );
        vips_error_exit( NULL );
    }
    g_object_unref( image );
    image = x;

    if( vips_avg( image, &mean, NULL ) ) {
        g_object_unref( image );
        vips_error_exit( NULL ); 
    }

    if( vips_deviate( image, &stddev, NULL ) ) {
        g_object_unref( image );
        vips_error_exit( NULL );
    }
    
    g_object_unref( image );

    printf( "mean = %g\n", mean );
    printf( "stddev = %g\n", stddev );

    vips_shutdown();

    return( 0 );
}
```
